compiler/src/dotty/tools/dotc/config/Settings.scala

```diff
@@ -88,7 +88,13 @@ object Settings {
         state.update(idx, x.asInstanceOf[T])
     }
 
-    def isDefaultIn(state: SettingsState) = valueIn(state) == default
+    def isDefaultIn(state: SettingsState): Boolean = valueIn(state) == default
+
+    def defaultValue: String = implicitly[ClassTag[T]] match {
+      case StringTag => default.asInstanceOf[String]
+      case IntTag => default.asInstanceOf[Int].toString
+      case _ => ""
```

Comment:
Both branches could be `default.toString`, but it seems you want to exclude defaults that aren't String and Int? Could you explain why?

Reply:
I found that only for these tag classes there are meaningful default values. For example for the `-version` there is `default: Boolean == false`, what does not make sense.